src/main/java/org/apache/sysml/hops/ipa/InterProceduralAnalysis.java

```diff
@@ -89,7 +89,7 @@
 	protected static final boolean INTRA_PROCEDURAL_ANALYSIS      = true; //propagate statistics across statement blocks (main/functions)	
 	protected static final boolean PROPAGATE_KNOWN_UDF_STATISTICS = true; //propagate statistics for known external functions 
 	protected static final boolean ALLOW_MULTIPLE_FUNCTION_CALLS  = true; //propagate consistent statistics from multiple calls 
-	protected static final boolean REMOVE_UNUSED_FUNCTIONS        = true; //remove unused functions (inlined or never called)
+	protected static final boolean REMOVE_UNUSED_FUNCTIONS        = false; //remove unused functions (inlined or never called)
 	protected static final boolean FLAG_FUNCTION_RECOMPILE_ONCE   = true; //flag functions which require recompilation inside a loop for full function recompile
```

Comment:
Once we merge this in, we would need to make this conditional on the presence of eval functions. I'm happy to take this over, but if you want to give it a shot, you can try extending the existing function call graph to capture this information and create the IPA rewrite set accordingly.